src/main/java/org/apache/sysml/api/mlcontext/MLContextUtil.java

```diff
@@ -585,24 +587,24 @@ public static boolean isCSVLineAllNumbers(String line) {
 	}
 
 	/**
-	 * If no metadata is supplied for a DataFrame, this method can be used to
-	 * determine whether the data appears to be a matrix (or a frame)
+	 * Examine the DataFrame schema to determine whether the data appears to be
+	 * a matrix.
 	 * 
-	 * @param row
-	 *            a row in the DataFrame
-	 * @return {@code true} if the row appears to be a matrix row, {@code false}
-	 *         otherwise
+	 * @param df
+	 *            the DataFrame
+	 * @return {@code true} if the DataFrame appears to be a matrix,
+	 *         {@code false} otherwise
 	 */
-	public static boolean doesRowLookLikeMatrixRow(Row row) {
-		for (int i = 0; i < row.length(); i++) {
-			Object object = row.get(i);
-			if (object instanceof Vector) {
-				return true;
-			}
-			String str = object.toString();
-			try {
-				Double.parseDouble(str);
-			} catch (NumberFormatException e) {
+	public static boolean doesDataFrameLookLikeMatrix(DataFrame df) {
+		StructType schema = df.schema();
+		StructField[] fields = schema.fields();
+		if (fields == null) {
+			return true;
+		}
+		for (StructField field : fields) {
+			DataType dataType = field.dataType();
+			if ((dataType != DataTypes.DoubleType) && (dataType != DataTypes.IntegerType)
```

Comment:
Confirmed that DataFrames with a `BigInt` type (i.e. a bigint for the `__INDEX`) will get converted correctly on the master branch.  Let's just add an additional `BigInt` type here.